### PR TITLE
Enh/add runner workflow

### PR DESCRIPTION
**Pull request by SebiWrn**
Adds a new workflow the the github workflows so runners from the beta branch can be deployed

### CODE REVIEW

File: .github/workflows/deploy-runners.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      matrix:
+        module: ["runner"]
+
+    steps:
+      - name: get container name
+        run: |
+          echo "containername=${{ matrix.module }}" | sed 's/\//-/g' > $GITHUB_ENV
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```
Comment (CommanderStorm) [inline .github/workflows/deploy-runners.yml:25]: - Let's use `checkout@v4` and
- add a comment wat this ref is for (I'd assume this or something equivalent should be the default, right?)

File: .github/workflows/deploy-runners.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set outputs
+        id: vars
+        run: echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
+      - name: Build and push Docker images (${{ matrix.module }})
+        uses: docker/build-push-action@v4
```
Comment (CommanderStorm) [inline .github/workflows/deploy-runners.yml:52]: You might want to configure sbom and build provence here.
Example how I did this:
https://github.com/TUM-Dev/NavigaTUM/blob/main/.github/workflows/_docker-build.yml

(This does not have an immediate impact but allows more forensics for some attack scenarios for little to no maintenance-effort as far as I read up on)

File: .github/workflows/deploy-runners.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        with:
+          context: ./${{ matrix.module }}
+          pull: true
+          push: true
+          build-args: version=${{ github.ref_name }}
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
```
Comment (CommanderStorm) [inline .github/workflows/deploy-runners.yml:59]: You might want to configure `cache-from` and related parameters.
Here is an example how to do this: https://docs.docker.com/build/ci/github-actions/cache/#cache-mounts

Likely the builds are already really fast (looking at you, rust), but a few seconds for a few lines...

**Review by CommanderStorm — APPROVED** [review]
LGTM, have provided some enhancement proposals but no bugs/.. ^^

**Comment by SebiWrn** [comment]
Thank you for the feedback, I think I'll deploy it anyway so we can deploy the beta system as fast as possible, but I'll keep your feedback in mind and open another PR for these changes so I can also change this on the current workers and tumlive deployment actions.

**Pull request merged**